### PR TITLE
Patch/credit card update route

### PR DESCRIPTION
**Pull request by ae-mo**
Related to #469.

This PR changes the credit card update functionality to be displayed full screen in a separate route.

It also introduces the `RouteWrapper` component which should help us dealing with displaying stuff full screen.

### CODE REVIEW

File: apps/mitt-konto/src/MittKonto/Main/Views/CreditCardUpdateView.purs [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+startRegisterPoller setState props state oldCreditCard creditCardRegister = do
+  newPoller <- Aff.forkAff do
+    killRegisterPoller state
+    newPoller <- Aff.forkAff $ pollRegister setState props state oldCreditCard (Right creditCardRegister)
+    Aff.joinFiber newPoller
+  liftEffect $ setState _ { poller = newPoller }
+
+pollRegister :: SetState -> Props -> State -> CreditCard -> Either BottegaError CreditCardRegister -> Aff Unit
+pollRegister setState props@{ logger, onError, onSuccess, onCancel } state oldCreditCard (Right register) = do
+  Aff.delay $ Aff.Milliseconds 1000.0
```
Comment (kaol) [inline apps/mitt-konto/src/MittKonto/Main/Views/CreditCardUpdateView.purs:163]: This doesn't feel like the right place for waiting. We know what the value of `register.status.state` is even before calling `delay`, why not act on it right away? If the value is `CreditCardRegisterCompleted` then won't this just add one more second to the whole process?

File: apps/mitt-konto/src/MittKonto/Wrappers/ViewWrapper.purs [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+  where
+    header :: JSX
+    header = Grid.row_
+      [ DOM.div
+          { className: "col col-11"
+          , children: [ title ]
+          }
+      , if closeable then
+          DOM.div
+            { className: "col-1 flex credit-card-choice--close-icon"
```
Comment (kaol) [inline apps/mitt-konto/src/MittKonto/Wrappers/ViewWrapper.purs:140]: I take it that `ViewWrapper.purs` is meant for other users besides credit cards? This class name wouldn't belong here then.

File: packages/components/src/Navbar/Component.purs [hunk 3]
```diff
@@ -12,10 +12,9 @@ import Effect (Effect)
 import KSF.Icons (papers)
```
Comment (kaol) [inline packages/components/src/Navbar/Component.purs:12]: I would leave pure style changes to a different pull request, this is large enough even without and they risk merge conflicts.

File: apps/mitt-konto/src/MittKonto/Wrappers/ViewWrapper.purs (hunk 2 above)
Comment (ae-mo) [reply apps/mitt-konto/src/MittKonto/Wrappers/ViewWrapper.purs:140]: Yes of course that is not right 😃

File: packages/components/src/Navbar/Component.purs (hunk 3 above)
Comment (ae-mo) [reply packages/components/src/Navbar/Component.purs:12]: 🤔  I don't think I have anything to do with this 😅

File: apps/mitt-konto/src/MittKonto/Main/Views/CreditCardUpdateView.purs (hunk 1 above)
Comment (ae-mo) [reply apps/mitt-konto/src/MittKonto/Main/Views/CreditCardUpdateView.purs:163]: It's a trick to poll every second, but we can think of a better place for that delay sure.

I never really questioned it because I started this component by copying **vetrina**'s code, but it is certainly good to scrutinize it, cause that delay is indeed annoying when the register operation has succeded.

**Comment by ae-mo** [comment]
The route of the credit card update view is currently `/kortt/uppdatera`, but I think there might be a better alternative 😅 I don't even know if it's correct Swedish

**Comment by vapaj** [comment]
> The route of the credit card update view is currently `/kortt/uppdatera`, but I think there might be a better alternative  I don't even know if it's correct Swedish

I think `/kort/uppdatera` (with one t) is close enough 😅 Or even `/kreditkort/uppdatera` to be precise.

File: packages/components/src/Navbar/Component.purs (hunk 3 above)
Comment (ae-mo) [reply packages/components/src/Navbar/Component.purs:12]: @kaol What did you exactly mean? I don't know how to act on this :blush:

Comment (kaol) [reply packages/components/src/Navbar/Component.purs:12]: Never mind, now that I looked a bit closer I saw that this was needed since `User` was added to User.purs. I first thought that this just rearranged imports to no effect.

File: apps/mitt-konto/src/MittKonto/Wrappers/RouteWrapper.purs [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+            { className: "route-wrapper"
+            , children:
+                [ header
+                , renderedContent
+                ] <> [ autoClose self.props closeAutomatically ]
+            }
+        }
+      where
+        header :: JSX
+        header = Grid.row_
```
Comment (vapaj) [inline apps/mitt-konto/src/MittKonto/Wrappers/RouteWrapper.purs:73]: I would not like to use the `Grid` module we have anymore in new code as it's dependent on [basscss](https://basscss.com/) which we don't need IMO. I think it's better to use plain css grids as they are really straight forward.

**Comment by vapaj** [comment]
Don't know where to exactly leave this comment to, but the `noMatchRoute` does not work anymore in this branch 🤔 So for instance `http://localhost:8000/asdasdasds` should render the mitt konto front page but it doesn't

File: apps/mitt-konto/src/MittKonto/Main/Views/CreditCardUpdateView.purs [hunk 5]
```diff
@@ -0,0 +1,194 @@
...
+didMount self@{ setState, props: { creditCards, logger, setWrapperState } } = do
+  setState _ { asyncWrapperState = AsyncWrapper.Ready }
+  Aff.launchAff_ do
+    case creditCards of
+      []       -> liftEffect $ do
+        logger.log "No credit cards found" Sentry.Error
+        onError self
+      [ card ] -> do
+        registerCreditCard self card
+      _        -> pure unit
```
Comment (vapaj) [inline apps/mitt-konto/src/MittKonto/Main/Views/CreditCardUpdateView.purs:86]: Should we take `Maybe CreditCard` (or do we even need `Maybe`?) in props instead of `Array CreditCard` if only one card makes sense in this component?

Comment (vapaj) [reply apps/mitt-konto/src/MittKonto/Main/Views/CreditCardUpdateView.purs:86]: Ohh, you can choose if there are multiple cards, but if there's only one, it's registered automatically?

Comment (ae-mo) [reply apps/mitt-konto/src/MittKonto/Main/Views/CreditCardUpdateView.purs:86]: > Ohh, you can choose if there are multiple cards, but if there's only one, it's registered automatically?

Yes that is the deal

**Comment by ae-mo** [comment]
> Don't know where to exactly leave this comment to, but the `noMatchRoute` does not work anymore in this branch 🤔 So for instance `http://localhost:8000/asdasdasds` should render the mitt konto front page but it doesn't

It does not work in prod either, I've checked. I think this is better suited for a new issue.

**Review by kaol — APPROVED** [review]
